Framework/MDAlgorithms/src/SlicingAlgorithm.cpp

```diff
@@ -1002,5 +1006,97 @@ SlicingAlgorithm::getImplicitFunctionForChunk(const size_t *const chunkMin,
+ */
+Mantid::Geometry::MDFrame_uptr SlicingAlgorithm::createMDFrameForNonAxisAligned(
+    std::string units, Mantid::Kernel::VMD basisVector) const {
+  // Get set of basis vectors
+  auto oldBasis = getOldBasis(m_inWS->getNumDims());
+
+  // Get indices onto which the vector projects
+  auto indicesWithProjection = getIndicesWithProjection(basisVector, oldBasis);
+
+  // Extract MDFrame
+  return extractMDFrameForNonAxisAligned(indicesWithProjection, units);
+}
+
+std::vector<Mantid::Kernel::VMD>
+SlicingAlgorithm::getOldBasis(size_t dimension) const {
+  std::vector<Mantid::Kernel::VMD> oldBasis;
+  for (size_t i = 0; i < dimension; ++i) {
+    Mantid::Kernel::VMD basisVector(dimension);
+    basisVector[i] = 1.0;
+    oldBasis.push_back(basisVector);
+  }
+  return oldBasis;
+}
+
+/**
+ * Check if the two vectors are orthogonal or not
+ * @param oldVector: the old vector
+ * @param basisVector: the vector under investigation
+ * @returns true if there is a projection  else false
+ */
+bool SlicingAlgorithm::isProjectingOnFrame(
+    const Mantid::Kernel::VMD &oldVector,
+    const Mantid::Kernel::VMD &basisVector) const {
+  return std::fabs(oldVector.scalar_prod(basisVector)) > 0.0;
+}
+
+/**
+ * Get indices which have a projection contribution
+ * @param basisVector: the vector under investigation
+ * @param oldBasis: the old basis vectors
+ * @returns the indices of vectors onto which the basisVector projects
+ */
+std::vector<size_t> SlicingAlgorithm::getIndicesWithProjection(
+    const Mantid::Kernel::VMD &basisVector,
+    const std::vector<Mantid::Kernel::VMD> &oldBasis) const {
+  std::vector<size_t> indexWithProjection;
+  for (size_t index = 0; index < oldBasis.size(); ++index) {
+    if (isProjectingOnFrame(oldBasis[index], basisVector)) {
+      indexWithProjection.push_back(index);
+    }
+  }
+  return indexWithProjection;
+}
+
+/**
+ * Extract the MDFrame. Make sure that all MDFrames are compatible -- if not
+ * throw
+ * @param indicesWithProjection: list of indices of dimensions which have a
+ * projection
+ * @param units: the units
+ */
+Mantid::Geometry::MDFrame_uptr
+SlicingAlgorithm::extractMDFrameForNonAxisAligned(
+    std::vector<size_t> indicesWithProjection, std::string) const {
+  if (indicesWithProjection.empty()) {
+    throw std::runtime_error("Slicing Algorithm: Chosen vector does not "
+                             "project on any vector of the old basis.");
+  }
+  // Get a reference frame to perform pairwise comparison
+  const auto &referenceMDFrame =
+      m_inWS->getDimension(indicesWithProjection[0])->getMDFrame();
+
+  for (auto it = indicesWithProjection.begin();
+       it != indicesWithProjection.end(); ++it) {
+    const auto &toCheckMDFrame = m_inWS->getDimension(*it)->getMDFrame();
+    if (!referenceMDFrame.isSameType(toCheckMDFrame)) {
+      throw std::runtime_error("Slicing Algorithm: New basis yvector tries to "
+                               "mix un-mixable MDFrame types.");
```

Comment:
I think this should be non-terminal. Users should see a warning. Just take the first frame of the overlapping. I'm worried that we've missed a use case.